packages/ckeditor5-ui/src/search/text/searchtextview.ts

```diff
@@ -0,0 +1,388 @@
+					primaryText = defaultTextConfig?.notFound?.primary || t( 'No results found' );
+					secondaryText = defaultTextConfig?.notFound?.secondary || '';
+				} else {
+					primaryText = defaultTextConfig?.noSearchableItems?.primary || t( 'No searchable items' );
+					secondaryText = defaultTextConfig?.noSearchableItems?.secondary || '';
+				}
+
+				infoView.set( {
+					primaryText: normalizeInfoText( primaryText, data ),
+					secondaryText: normalizeInfoText( secondaryText, data ),
+					isVisible: true
+				} );
+			} else {
+				infoView.set( {
+					isVisible: false
+				} );
+			}
+		} );
+
+		function normalizeInfoText(
+			text: SearchTextViewDefaultInfoText,
+			{ query, resultsCount, totalItemsCount }: SearchTextViewSearchEvent[ 'args' ][ 0 ]
+		) {
+			return typeof text === 'function' ? text( query, resultsCount, totalItemsCount ) : text;
+		}
+	}
+}
+
+/**
+ * The configuration of the {@link module:ui/search/text/searchtextview~SearchTextView} class.
+ */
+export interface SearchTextViewConfig<TConfigSearchField extends InputBase<HTMLInputElement | HTMLTextAreaElement>> {
+
+	/**
+	 * The configuration of the view's query field.
+	 */
+	queryView: SearchTextQueryViewConfig<TConfigSearchField>;
+
+	/**
+	 * The view that is filtered by the search query.
+	 */
+	filteredView: FilteredView;
+
+	/**
+	 * The view that displays the information about the search results.
+	 */
+	infoView?: {
+
+		/**
+		 * The view that displays the information about the search results. If not specified,
+		 * {@link module:ui/search/searchinfoview~SearchInfoView} is used.
+		 */
+		instance?: View;
+
+		/**
+		 * The configuration of text labels displayed in the {@link #infoView} in different states
+		 * of the search component.
+		 *
+		 * **Note**: This configuration is only used when the {@link #infoView} is **not** specified.
+		 * In other cases, please use the {@link module:ui/search/searchview~SearchTextViewSearchEvent} to bring about
+		 * your own info text logic.
+		 */
+		text?: {
+			notFound?: {
+				primary: SearchTextViewDefaultInfoText;
+				secondary?: SearchTextViewDefaultInfoText;
+			};
+			noSearchableItems?: {
+				primary: SearchTextViewDefaultInfoText;
+				secondary?: SearchTextViewDefaultInfoText;
+			};
+		};
+	};
+
+	/**
+	 * The custom CSS class name to be added to the search view element.
+	 */
+	class?: string;
+}
+
+/**
+ * Describes value of a info text configuration in {@link module:ui/search/text/searchtextview~SearchTextViewConfig}.
+ * A string or a function that returns a string with the information about the search results.
+ */
+export type SearchTextViewDefaultInfoText = string | ( ( query: string, resultsCount: number, totalItemsCount: number ) => string );
+
+/**
+ * An event fired when the search query changes fired by {@link module:ui/search/text/searchtextview~SearchTextView#search}.
+ *
+ * @eventName ~SearchTextView#search
+ */
+export type SearchTextViewSearchEvent = {
+	name: 'search';
+	args: [ {
```

Comment:
I'm pretty sure, API Docs for this will not be useful. Consider extracting this object type as a separate (and documented) `interface`. If you think this is too much, at least mention the fields in the event description (the doclet above)).